src/select.ts

```diff
@@ -114,7 +114,7 @@ export class PrismaSelect {
   }
 
   static isObject(item: any) {
-    return item && typeof item === 'object' && !Array.isArray(item)
+    return item && typeof item === 'object' && !Array.isArray(item) && typeof item?.toISOString !== 'function'
```

Comment:
Shouldn't we explicitly test for `Date`?

```suggestion
    return item && typeof item === 'object' && !Array.isArray(item) && !(item instanceof Date)
```

Reply:
nah. that's a bad idea. `Date` often is not the same things across various context boundaries. Since this is a thrid-party lib it's a better practice to duck-type it...

Reply:
But many things can expose a `toISOString` no? Not just `Date`ish things.

Which context boundaries wouldn't have `Date` be a _date_?

Reply:
e.g. in frontend it would be frame boundaries. serverside I cannot come up with an example, but who knows.
we can replace the check with `Object.prototype.toString.call(date) === "[object Date]"` if you prefer.

Reply:
> serverside I cannot come up with an example, but who knows.

I mean this is the core of my wondering – why are we being this defensive?

> `Object.prototype.toString.call(date) === "[object Date]"` if you prefer.

This is certainly clearer to me. Still odd though :shrug:

Reply:
>  why are we being this defensive?

Because it's the best practice to not trust object prototypes to have referential integrity.
But not sure who's being more defensive, me with not trusting prototypes or you expecting value objects to have `toISOString` :-D